bubuku/features/restart_if_dead.py

```diff
@@ -18,18 +19,31 @@ def __init__(self, broker: BrokerManager, zk: BukuExhibitor):
     def check(self) -> Change:
         if not self.need_check:
             return None
-        if self.broker.is_running_and_registered():
+        if self.is_running_and_registered(tries=3):
             return None
 
         _LOG.info('Oops! Broker is dead, triggering restart')
         self.need_check = False
 
         # Do not start if broker is running and registered
         def _cancel_if():
-            return self.broker.is_running_and_registered()
+            return self.is_running_and_registered()
 
         return RestartBrokerChange(self.zk, self.broker, _cancel_if, self.on_check_removed)
 
+    def is_running_and_registered(self, tries=1):
+        if not self.broker.process.is_running():
```

Comment:
It is better to create 2 methods in broker class: `is_process_running()` and `is_broker_registered()`, so internals of broker (that are incorrectly named without underscore) wont' leak anywhere

Reply:
Agreed